### PR TITLE
Issue #4936: Replace multiple occurences of `config_get()` in the same core functions with `$config->get`

### LINKED ISSUE #4936: Replace multiple occurences of `config_get()` in the same core functions with `$config->get` implementation
**Description of the bug**

As I mentioned in https://github.com/backdrop/backdrop-issues/issues/4664:

> I haven't analyzed other files, so there could be some other cases of mixed use of two methods in the same core functions. The above case is just what I've noticed in the system.admin.inc while working on something else, so I've decided to create a standalone issue for this particular file (system.admin.inc) only.

Checking other core files revealed plenty use of `config_get()` within the same functions, that is not ideal, because per https://docs.backdropcms.org/api/backdrop/core%21includes%21config.inc/function/config_get/1 the function:

```
function config_get($config_file, $option = NULL) {
  $config = config($config_file);
  return $config->get($option);
}
```

already reads the configuration file with `config()`, so there is no need to do it again.

**Additional information**

I've left all the instances of `config_get()` in mutually exclusive condition checks like:

```
function date_default_timezone($check_user = TRUE) {
  global $user;
  if ($check_user && config_get('system.date','user_configurable_timezones') && !empty($user->timezone)) {
    return $user->timezone;
  }
  else {
    $default = config_get('system.date','default_timezone');
    return empty($default) ? 'UTC' : $default;
  }
}
```

because thanks to `if else` it's always one configuration read. 

At the same time, wherever there is potentially two ore more reads, I've replaced with the recommended on https://docs.backdropcms.org/change-records/variableget-variableset-and-variabledel-deprecated-cmi method:

> Every time you call config_get() or config_set(), it hits the disk to read/write the file. If you're doing a lot of reads or writes, it's better to do all the reads at once and all the writes at once like this:
> 
> ```
> $config = config('mymodule.settings');
> $value = $config->get('foo');
> ```

### PR DESCRIPTION
**Pull request by alanmels**
Fixes https://github.com/backdrop/backdrop-issues/issues/4936

### CODE REVIEW

File: core/includes/common.inc [hunk 1]
```diff
@@ -3019,14 +3021,13 @@ function backdrop_deliver_html_page($page_callback_result) {
   // domain, but provide a variable to override this. If the code running for
   // this page request already set the X-Frame-Options header earlier, don't
   // overwrite it here.
-  $frame_options = config_get('system.core', 'x_frame_options');
+  $frame_options = config_get('system.core','x_frame_options');
```
Comment (klonos) [inline core/includes/common.inc]: This change here should be reverted:

```suggestion
  $frame_options = config_get('system.core', 'x_frame_options');
```

Comment (klonos) [reply core/includes/common.inc]: ...also, I believe that the intention was to use:

```suggestion
  $frame_options = $site_config->get('x_frame_options');
```

File: core/modules/path/path.admin.inc [hunk 2]
```diff
@@ -123,14 +123,15 @@ function path_admin_edit($path = array()) {
...
+  $site_config = config('system.core');
   $form['source'] = array(
     '#type' => 'textfield',
     '#title' => t('Existing system path'),
     '#default_value' => $path['source'],
     '#maxlength' => 255,
     '#size' => 45,
     '#description' => t('Specify the existing path you wish to alias. For example: <code>node/28</code> or <code>taxonomy/term/1</code>.'),
-    '#field_prefix' => url(NULL, array('absolute' => TRUE)) . (config_get('system.core', 'clean_url') ? '' : '?q='),
+    '#field_prefix' => url(NULL, array('absolute' => TRUE)) . ($site_config->get('clean_url') ? '' : '?q='),
```
Comment (klonos) [inline core/modules/path/path.admin.inc]: Here, we could do something like:

```php
  $field_prefix = url(NULL, array('absolute' => TRUE)) . ($site_config->get('clean_url') ? '' : '?q=');
```

...and then reuse:

```php
  '#field_prefix' => $field_prefix,
```

It would save us 2 calls to `url()` 😉 ...but perhaps scope creep for this PR.

Comment (alanmels) [reply core/modules/path/path.admin.inc]: Nice catch, Greg. How about if we leave this to a standalone PR, because if the review will be elevated to someone else, I don't want to confuse this PR with any code change other then just replacing `config_get`s with `site_config->get`s.

Comment (alanmels) [reply core/modules/path/path.admin.inc]: But the other excellent catch of yours I have already committed. Thank you for noticing it!

File: core/includes/menu.inc [hunk 3]
```diff
@@ -539,8 +539,9 @@ function menu_execute_active_handler($path = NULL, $deliver = TRUE, $route_handl
   // Offline pages are not cached in the database but have a Cache-Control
   // header to allow reverse proxies to serve offline hits instead of Backdrop.
   elseif ($site_status == MENU_SITE_OFFLINE) {
-    $page_cache_enabled = (bool) config_get('system.core', 'page_cache_maximum_age');
-    $maintenance_max_age = config_get('system.core', 'maintenance_page_maximum_age');
+    $site_config = config('system.core');
```
Comment (indigoxela) [inline core/includes/menu.inc]: You could even move that up to the beginning of function menu_execute_active_handler(). - In line 524 there's already a call to `config_get('system.core', 'menu_route_handler');`

File: core/includes/theme.maintenance.inc [hunk 4]
```diff
@@ -15,6 +15,7 @@
  */
 function _backdrop_maintenance_theme() {
   global $theme, $theme_key;
+  $site_config = config('system.core');
```
Comment (indigoxela) [inline core/includes/theme.maintenance.inc]: In this function it might make sense to move that config() down into the try{} - as that's the only place where this config is used.

There are two situations, where this config isn't needed at all, so it could be better to call it only if it's actually used.

File: core/modules/installer/installer.manager.inc [hunk 5]
```diff
@@ -59,6 +59,7 @@
  * @ingroup forms
  */
 function installer_manager_update_form($form, $form_state, $context) {
+  $installer_settings_config = config('installer.settings');
```
Comment (indigoxela) [inline core/modules/installer/installer.manager.inc]: Actually, in this loooong function :smile:  it's not only the same config, but also the same config setting: `config_get('installer.settings', 'core_update')`

It this case it would be possible to set a variable and use that twice, for instance:
```
$core_update_enabled = config_get('installer.settings', 'core_update');
```
It's up to you, whatever you think is best.

File: core/modules/path/path.admin.inc [hunk 6]
```diff
@@ -123,14 +123,15 @@ function path_admin_edit($path = array()) {
  * @see path_admin_form_delete_submit()
  */
 function path_admin_form($form, &$form_state, $path = array('source' => '', 'alias' => '', 'langcode' => LANGUAGE_NONE, 'pid' => NULL)) {
+  $site_config = config('system.core');
```
Comment (indigoxela) [inline core/modules/path/path.admin.inc]: Both occurrences of config_get() actually use the same config item, so it would be possible to:
```
$clean_urls = config_get('system.core', 'clean_url');
```
And use this variable in both places. There's no other config item used in this function.

File: core/modules/system/system.install [hunk 7]
```diff
@@ -9,6 +9,7 @@
  */
 function system_requirements($phase) {
   global $base_url;
+  $site_config = config('system.core');
```
Comment (indigoxela) [inline core/modules/system/system.install:12]: It might make sense to move this down to the phase and condition where it's actually needed - `if ($phase == 'runtime' && backdrop_is_apache()) `

Up here it's called, although it might not be necessary anyway, and is more than 200 lines away from its actual usage (line 254, 256).

Comment (indigoxela) [reply core/modules/system/system.install:12]: There are several more places where it's in use, but again so far away, that it could be more readable, if it got set down there in these multiple `if ($phase == 'runtime') {`

My concern here is readability. In line 606 `$site_config` comes "out of nothing"... Per se your change is correct, though, so I'll leave it to you to decide.

File: core/modules/translation/tests/translation.test [hunk 8]
```diff
@@ -417,6 +417,7 @@ class TranslationTestCase extends BackdropWebTestCase {
    *   TRUE if the language switch links are found, FALSE if not.
    */
   function assertLanguageSwitchLinks($node, $translation, $find = TRUE, $types = NULL) {
+    $site_config = config('system.core');
```
Comment (indigoxela) [inline core/modules/translation/tests/translation.test]: Again... :wink:  ... or you set a variable like:
```
$language_default = config_get('system.core', 'language_default');
```
As this language_default is the only config item needed here.

File: core/modules/user/user.module [hunk 9]
```diff
@@ -3186,9 +3189,10 @@ function user_register_validate($form, &$form_state) {
  * @see user_register_form()
  */
 function user_register_submit($form, &$form_state) {
+  $site_config = config('system.core');
```
Comment (indigoxela) [inline core/modules/user/user.module]: Hm... it seems, I repeat myself... :laughing: 

As 'user_email_verification' is the only thing needed in this function, you could:
```
$user_email_verification = config_get('system.core', 'user_email_verification');
```
And use that variable later on.

File: core/modules/user/user.tokens.inc [hunk 10]
```diff
@@ -96,6 +96,7 @@ function user_token_info() {
  * Implements hook_tokens().
  */
 function user_tokens($type, $tokens, array $data = array(), array $options = array()) {
+  $site_config = config('system.core');
```
Comment (indigoxela) [inline core/modules/user/user.tokens.inc]: Oh, I really repeat myself often... I hope, it doesn't get boring.
```
$user_pictures = config_get('system.core', 'user_pictures');
```
Only "user_pictures" is needed in this function.

File: core/modules/views/includes/view.inc [hunk 11]
```diff
@@ -1224,6 +1224,7 @@ class view {
    */
   function render($display_id = NULL) {
     $this->execute($display_id);
+    $views_settings_config = config('views.settings');
```
Comment (indigoxela) [inline core/modules/views/includes/view.inc]: ... ehm ... guess what:
```
$show_additional_queries = config_get('views.settings', 'show_additional_queries');
```
Only "show_additional_queries" is needed in this function.

File: core/includes/menu.inc (hunk 3 above)
Comment (alanmels) [reply core/includes/menu.inc]: Nice catch! Thanks for the suggestion. Committed.

File: core/includes/theme.maintenance.inc (hunk 4 above)
Comment (alanmels) [reply core/includes/theme.maintenance.inc]: As always you are right, @indigoxela! Learning from you!

File: core/modules/installer/installer.manager.inc (hunk 5 above)
Comment (alanmels) [reply core/modules/installer/installer.manager.inc]: Followed your recommendation and changed the file.

File: core/modules/system/system.install (hunk 7 above)
Comment (alanmels) [reply core/modules/system/system.install:12]: I had already started to make the recommended changes, when I noticed there are multiple instances of `if ($phase == 'runtime') {` condition. Which means if we move config read inside of `if` condition, then we would have multiple reads. So let's leave the config read in the very beginning of the function and re-use $config object anywhere we need.

File: core/modules/translation/tests/translation.test (hunk 8 above)
Comment (alanmels) [reply core/modules/translation/tests/translation.test]: Thanks for noting this one too. Changed.

File: core/modules/user/user.module (hunk 9 above)
Comment (alanmels) [reply core/modules/user/user.module]: Done!

File: core/modules/user/user.tokens.inc (hunk 10 above)
Comment (alanmels) [reply core/modules/user/user.tokens.inc]: Not boring at all, it's enlightening, @indigoxela! Thanks a lot!

File: core/modules/views/includes/view.inc (hunk 11 above)
Comment (alanmels) [reply core/modules/views/includes/view.inc]: Absolutely right! Corrected.

**Review by alanmels — COMMENTED** [review]
I think I've done all the recommended changes. Let me know if you think something else should be changed too.

File: core/modules/path/path.admin.inc (hunk 6 above)
Comment (alanmels) [reply core/modules/path/path.admin.inc]: Done!

**Review by alanmels — COMMENTED** [review]
I hope this is all.

File: core/includes/common.inc (hunk 1 above)
Comment (alanmels) [reply core/includes/common.inc]: Thanks for the catch, @klonos. Committed.

**Comment by quicksketch** [comment]
Merged into 1.x and 1.18.x. Thanks!
